tools/clang/unittests/HLSL/ShaderOpArithTable.xml

```diff
@@ -2388,4 +2388,689 @@
+                            uint one = 1;
+                            PerPixelData d;
+                            d.position = input.position;
+                            d.id = pos_to_id(input.position);
+                            d.flags = 0;
+                            if (WaveIsFirstLane()) d.flags |= 1;
+                            d.laneIndex = WaveGetLaneIndex();
+                            d.laneCount = WaveGetLaneCount();
+                            d.firstLaneId = WaveReadLaneFirst(d.id);
+                            d.sum1 = WaveActiveSum(one);
+                            d.id0 = QuadReadLaneAt(d.id, 0);
+                            d.id1 = QuadReadLaneAt(d.id, 1);
+                            d.id2 = QuadReadLaneAt(d.id, 2);
+                            d.id3 = QuadReadLaneAt(d.id, 3);
+                            d.acrossX = QuadReadAcrossX(d.id);
+                            d.acrossY = QuadReadAcrossY(d.id);
+                            d.acrossDiag = QuadReadAcrossDiagonal(d.id);
+                            d.quadActiveCount = one + QuadReadAcrossX(one) + QuadReadAcrossY(one) + QuadReadAcrossDiagonal(one);
+                            g_sb.Append(d);
+                            return 1;
+                        }
+                ]]>
+            </Parameter>
+            <Parameter Name="Validation.DataType"></Parameter>
+            <Parameter Name="Validation.Input"></Parameter>
+            <Parameter Name="Validation.Output"></Parameter>
+        </Row>
+
+       </Table>
+
+       <Table Id="WaveIntrinsicsActiveIntTable">
+        <ParameterTypes>
+            <ParameterType Name="ShaderOp.Name">String</ParameterType>
+            <ParameterType Name="ShaderOp.Text">String</ParameterType>
+            <ParameterType Name="Validation.NumInputSet">unsigned int</ParameterType>
+            <ParameterType Name="Validation.InputSet1" Array="true">int</ParameterType>
+            <ParameterType Name="Validation.InputSet2" Array="true">int</ParameterType>
+            <ParameterType Name="Validation.InputSet3" Array="true">int</ParameterType>
+            <ParameterType Name="Validation.InputSet4" Array="true">int</ParameterType>
+        </ParameterTypes>
+
+        <Row Name="WaveActiveSum">
+            <Parameter Name="ShaderOp.Name">WaveActiveSum</Parameter>
+            <Parameter Name="ShaderOp.Text">
+                <![CDATA[
+                    struct PerThreadData {
+                        int id;
+                        int firstLaneId;
+                        int input;
+                        int output;
+                    };
+                    RWStructuredBuffer<PerThreadData> g_sb : register(u0);
+                    [numthreads(8,8,1)]
+                    void main(uint GI : SV_GroupIndex) {
+                        PerThreadData pts = g_sb[GI];
+                        pts.firstLaneId = WaveReadLaneFirst(pts.id);
+                        pts.output = WaveActiveSum(pts.input);

```

Comment:
Let's add some flow control so not all waves are active.  (same with others)
Suggestion: if (pts.active) with new parameter table for this.